module/input/SpeechIntent/src/SpeechIntent.cpp

```diff
@@ -0,0 +1,425 @@
+                              "-p",
+                              "en",
+                              "transcribe-stream",
+                              NULL};
+
+        return spawn_process(process, args);
+    }
+
+    // python -m voice2json --base-directory /home/nubots/voice2json/ -p en transcribe-wav --input-size
+    bool voice2json_transcribe_wav(SpawnedProcess& process) {
+        const char* args[] = {"python",
+                              "-m",
+                              "voice2json",
+                              "--base-directory",
+                              "/home/nubots/voice2json/",
+                              "-p",
+                              "en",
+                              "transcribe-wav",
+                              "--input-size",
+                              NULL};
+        return spawn_process(process, args);
+    }
+
+    // python -m voice2json --base-directory /home/nubots/voice2json/ -p en recognize-intent
+    char* voice2json_recognize_intent(char* input) {
+        char* output = 0;
+        char buffer[0x1000];
+        ssize_t bytes_read = 0;
+
+        const char* args[] = {"python",
+                              "-m",
+                              "voice2json",
+                              "--base-directory",
+                              "/home/nubots/voice2json/",
+                              "-p",
+                              "en",
+                              "recognize-intent",
+                              NULL};
+
+        SpawnedProcess process = {};
+        if (spawn_process(process, args)) {
+            write(process.stdin, input, strlen(input));
+
+            bytes_read = read(process.stdout, buffer, sizeof(buffer) - 1);
+            if (bytes_read == -1) {
+                NUClear::log<NUClear::FATAL>(
+                    fmt::format("({}:{}) failed to read from spawned process "
+                                "(voice2json recognize-intent) stdout - errno = {}",
+                                __FILE__,
+                                __LINE__,
+                                errno));
+            }
+            buffer[bytes_read] = 0;
+
+            output = strdup(buffer);
+        }
+        if (process.stdout) {
+            close(process.stdout);
+        }
+        if (process.stderr) {
+            close(process.stderr);
+        }
+        if (process.stdin) {
+            close(process.stdin);
+        }
+        return output;
+    }
+
+    // write out the correct format for voice2json to consume
+    // when voice2json is called using the "transcribe-wav --input-size" flags
+    // it excepts the first line to be the file size (as ascii)
+    // followed by the wav file after the newline
+    bool write_audio_to_file(int fd, char* filename) {
+        ssize_t write_res = 0;
+        if (access(filename, F_OK) != 0) {
+            return false;
+        }
+
+        FILE* file = fopen(filename, "rb");
```

Comment:
Try to avoid using the C standard library, C++ file ops should always be preferred